sisyphus/manager.py

```diff
@@ -593,7 +603,9 @@ def run(self):
             self.check_output(write_output=self.link_outputs)
 
             config_manager.continue_readers()
-            self.update_jobs()
+
+            prev_jobs = self.jobs
+            cur_jobs = self.update_jobs()
```

Comment:
How will this interact with the block directly below. `self.clear_states` removes some errors, call `self.update_jobs()`  and then cycle the outer loop again. This would mess up the logic to detect new jobs in error state.